File: frontend/src/types/review-summary.ts
export type ReviewRiskLevel = 'low' | 'medium' | 'high';

export interface FileReviewInsight {
  path: string;
  status: string;
  riskLevel: ReviewRiskLevel | string;
  reasons: string[];
  additions: number;
  deletions: number;
}

export interface WorkspaceReviewSummary {
  workspaceId: string;
  summary: string;
  riskLevel: ReviewRiskLevel | string;
  riskReasons: string[];
  filesChanged: number;
  filesFlagged: number;
  additions: number;
  deletions: number;
  generatedAt: string;
  fileInsights: FileReviewInsight[];
}

File: crates/mnemonic-core/src/models/review_summary.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct FileReviewInsight {
    pub path: String,
    pub status: String,
    pub risk_level: String,
    pub reasons: Vec<String>,
    pub additions: u32,
    pub deletions: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceReviewSummary {
    pub workspace_id: String,
    pub summary: String,
    pub risk_level: String,
    pub risk_reasons: Vec<String>,
    pub files_changed: u32,
    pub files_flagged: u32,
    pub additions: u32,
    pub deletions: u32,
    pub generated_at: String,
    pub file_insights: Vec<FileReviewInsight>,
}
